stdlib/database/sql/src/main/java/org/ballerinalang/sql/datasource/SQLDatasource.java

```diff
@@ -45,13 +48,30 @@
     private AtomicInteger clientCounter = new AtomicInteger(0);
     private Lock mutex = new ReentrantLock();
     private boolean poolShutdown = false;
+    private boolean xaConn;
+    private XADataSource xaDataSource;
 
     private SQLDatasource(SQLDatasourceParams sqlDatasourceParams) {
         buildDataSource(sqlDatasourceParams);
-        try (Connection con = getSQLConnection()) {
+        Connection connection = null;
+        try {
+            xaConn = hikariDataSource.isWrapperFor(XADataSource.class);
+            if (xaConn) {
+                xaDataSource = hikariDataSource.unwrap(XADataSource.class);
+                connection = xaDataSource.getXAConnection().getConnection();
+            } else {
+                connection = getConnection();
```

Comment:
Shouldn't we set autoCommit to false here? Else we cant rollback if an error occurs

Reply:
Here I'm just checking whether valid connection can be obtained, not doing any other queries. But for the case of transaction, I'm setting autoCommit to false as you mentioned here - https://github.com/ballerina-platform/ballerina-lang/blob/8d83edd04c02621b84629eb75d70081abb2ddd50/stdlib/database/sql/src/main/java/org/ballerinalang/sql/datasource/SQLDatasourceUtils.java#L112

Reply:
Noted